### ISSUE TITLE
ImportError: No module named ryu.lib.packet

### ISSUE THREAD

**Issue by zfloat** [frenetic-lang/frenetic#655]
I am now able to use French for simple network orchestration, but I still generate errors when compiling the learning. py file. Could you please help me solve this problem?

![图片](https://user-images.githubusercontent.com/63533219/230154558-559d19d6-4fab-49e7-aec7-11c939a30715.png)

**Comment by jnfoster** [comment]
I think you want `frenetic.examples.repeater`.

**Comment by zfloat** [comment]
I am now able to use French for simple network orchestration, but I still generate errors when compiling the learning. py file. Could you please help me solve this problem?

![图片](https://user-images.githubusercontent.com/63533219/230154700-4343f562-60fb-4bcb-b397-7d481703e6ae.png)

**Comment by jnfoster** [comment]
Oops. 

Install Ryu like this,
```
pip install ryu
```
and then try again?

**Comment by zfloat** [comment]
I have tried installing ryu using the code `pip install ryu`, but an error occurred during the installation process, as shown in the following figure.
![图片](https://user-images.githubusercontent.com/63533219/230282222-4e05ed4b-7e1c-47e5-ba41-7460ab763bb8.png)
This may mean that installing ryu requires Python 3, so in the end, I had to use `pip3 to install ryu`. Although ryu was successfully installed, errors will still be reported when compiling the learning. py file with Python 2.
I'm at a loss what to do next.

**Comment by jnfoster** [comment]
I think you need to use the version of Ryu with Python 2.7.

Maybe try the from-source instructions here?
https://osrg.github.io/ryu-book/en/html/installation_guide.html

**Comment by zfloat** [comment]
I have successfully solved the above problem by installing version 2.14.0 of Ovs before installing ryu. This is because the latest version of Ovs will generate errors when installed in the Python 2 environment.

Recently, I have been following the guide manual to further learn and use frenetic, but I have encountered errors when compiling the vlan2. py file. How can I solve this problem? 
The following is my experimental process.
`Copy the network_information_base_dynamic.py, network_information_base_static.py, and vlan2.py files into the examples folder.`
`frenetic http-controller --verbosity debug`
`sudo python2 mn_custom_topo.py `
`python2 -m frenetic.examples.vlan2`

At this point, compilation generates the following error.
![图片](https://user-images.githubusercontent.com/63533219/234221320-8e4f566f-f6e9-432b-889f-8c4a14426e86.png)

**Comment by jnfoster** [comment]
I assume this is from Chapter 5 of the manual. Can you share the content of `mn_custom_topo.py`?

**Comment by zfloat** [comment]
Yes, this is indeed the content of Chapter 5 of the manual. The mn_custom_topo.py file was also obtained from https://github.com/frenetic-lang/manual/blob/master/programmers_guide/code/handling_vlans/mn_custom_topo.py

After I run this topology file, the following information is displayed.
`sudo python2 mn_custom_topo.py `
![图片](https://user-images.githubusercontent.com/63533219/234498603-6deedc30-c167-4747-b567-18292a43f6ef.png)

**Comment by jnfoster** [comment]
This seems to be a bug with the way VLAN'd packets are being created. I'm afraid I don't have the cycles to debug it now. I mention that in case you want to dig in. The Python code *should* create packets on the right VLAN. But apparently it's not. You could use TCPdump or Wireshark to see what packets are hitting the wire, and then debug from there.
